src/test/java/bio/terra/common/fixtures/ConnectedOperations.java

```diff
@@ -458,6 +463,55 @@ public FileModel ingestFileSuccess(String datasetId, FileLoadModel fileLoadModel
         MockHttpServletResponse response = validateJobModelAndWait(result);
 
         FileModel fileModel = handleSuccessCase(response, FileModel.class);
+        checkSuccessfulFileLoad(fileLoadModel, fileModel);
+
+        logger.info("addFile datasetId:{} objectId:{}", datasetId, fileModel.getFileId());
+        addFile(datasetId, fileModel.getFileId());
+
+        return fileModel;
+    }
+
+    public void retryAcquireLockIngestFileSuccess(
+        boolean attemptRetry,
+        String datasetId,
+        FileLoadModel fileLoadModel,
+        ConfigurationService configService,
+        DatasetDao datasetDao) throws Exception {
+        // Insert fault into shared lock
+        ConfigEnum faultToInsert = attemptRetry ?
+            ConfigEnum.FILE_INGEST_SHARED_LOCK_RETRY_FAULT :
+            ConfigEnum.FILE_INGEST_SHARED_LOCK_FATAL_FAULT;
+        configService.setFault(faultToInsert.name(), true);
+
+        String jsonRequest = TestUtils.mapToJson(fileLoadModel);
+        String url = "/api/repository/v1/datasets/" + datasetId + "/files";
+        MvcResult result = mvc.perform(post(url)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(jsonRequest))
+            .andReturn();
+
+        TimeUnit.SECONDS.sleep(5); // give the flight time to fail a couple of times
+        datasetDaoUtils = new DatasetDaoUtils();
+        String[] sharedLocks = datasetDaoUtils.getSharedLocks(datasetDao, UUID.fromString(datasetId));
+
+        // Remove insertion of shared lock fault
+        configService.setFault(faultToInsert.name(), false);
+        // ====================================================
```

Comment:
Is there a specific reason for the emphasis in this section?

Reply:
Nope, good catch.